modvendor/modvendor.go

```diff
@@ -15,59 +15,220 @@
+		Str("module.source", modsrc.Raw).
+		Logger()
+
+	moddir, err := doVendor(rootdir, vendorDir, modsrc)
+	if err != nil {
+		if errors.IsKind(err, ErrAlreadyVendored) {
+			// it's not an error in the case it's an indirect vendoring
+			if info == nil {
+				report.addIgnored(modsrc.Raw, string(ErrAlreadyVendored))
+			}
+			return report
+		}
+
+		reason := errors.E(err, "failed to vendor %q with ref %q",
+			modsrc.URL, modsrc.Ref).Error()
+
+		if info != nil {
+			reason += fmt.Sprintf(" found in %s", info.origin)
+		}
+
+		report.addIgnored(modsrc.Raw, reason)
+		return report
+	}
+
+	logger.Trace().Msg("successfully downloaded")
+
+	report.addVendored(modsrc.Raw, modsrc, Dir(vendorDir, modsrc))
+	return vendorAll(rootdir, vendorDir, moddir, report)
+}
+
+func vendorAll(rootdir string, vendorDir string, tfdir string, report Report) Report {
 	logger := log.With().
-		Str("action", "modvendor.Vendor()").
+		Str("action", "modvendor.vendorAll()").
+		Str("dir", tfdir).
+		Logger()
+
+	logger.Trace().Msg("scanning .tf files for additional dependencies")
+
+	sourcemap := map[string]*modinfo{}
+	originMap := map[string]struct{}{}
+	errs := errors.L(report.Error)
+	err := filepath.WalkDir(tfdir, func(path string, d iofs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() && filepath.Base(path) == ".git" {
+			return filepath.SkipDir
+		}
+
+		if !d.Type().IsRegular() || !strings.HasSuffix(path, ".tf") {
+			return nil
+		}
+
+		logger.Trace().Str("path", path).Msg("found Terraform file")
+
+		modules, err := tf.ParseModules(path)
+		if err != nil {
+			errs.Append(err)
+			return nil
+		}
+
+		for _, mod := range modules {
+			logger = logger.With().
+				Str("module.source", mod.Source).
+				Logger()
+
+			if mod.IsLocal() {
+				logger.Trace().Msg("ignoring local module")
+				continue
+			}
+
+			logger.Trace().Msg("found remote module")
+
+			originMap[path] = struct{}{}
+			sourcemap[mod.Source] = &modinfo{
+				source: mod.Source,
+				origin: path,
+			}
+		}
+		return nil
+	})
+
+	errs.Append(err)
+
+	for source, info := range sourcemap {
+		logger := logger.With().
+			Str("module.source", source).
+			Str("origin", info.origin).
+			Logger()
+
+		if _, ok := report.Vendored[source]; ok {
+			logger.Trace().Msg("already vendored")
+			continue
+		}
+
+		modsrc, err := tf.ParseSource(source)
+		if err != nil {
+			report.addIgnored(source, err.Error())
+			delete(sourcemap, source)
+			continue
+		}
+		report = recVendor(rootdir, vendorDir, modsrc, report, info)
+		if _, ok := report.Vendored[source]; ok {
+			info.vendoredAt = Dir(vendorDir, modsrc)
+
+			logger.Trace().Msg("vendored successfully")
+		}
+	}
+
+	files := []string{}
+	for fname := range originMap {
+		files = append(files, fname)
+	}
+	sort.Strings(files)
+	errs.Append(patchFiles(rootdir, files, sourcemap))
+	report.Error = errs.AsError()
+	return report
+}
+
+// doVendor will doVendor the provided modsrc into the rootdir.
```

Comment:
```suggestion
// downloadVendor will download the provided modsrc into the vendorDir.
```

Just testing the new name, seems better =P